internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/TestPushJobWithNativeReplication.java

```diff
@@ -943,6 +926,25 @@ public void testTargetedRegionPushJob() throws Exception {
               }
             });
           }
+
+          // emergency source is dc-0 so dc-1 isn't selected to be the source fabric but the push should still complete
+          props.setProperty(TARGETED_REGION_PUSH_LIST, "dc-1");
+          try (VenicePushJob job = new VenicePushJob("Test push job 4", props)) {
+            job.run(); // the job should succeed
+
+            TestUtils.waitForNonDeterministicAssertion(30, TimeUnit.SECONDS, () -> {
+              Map<String, Integer> coloVersions =
+                  parentControllerClient.getStore(storeName).getStore().getColoToCurrentVersions();
+
+              coloVersions.forEach((colo, version) -> {
+                if (colo.equals(DEFAULT_NATIVE_REPLICATION_SOURCE)) {
+                  Assert.assertEquals((int) version, 3);
+                } else {
+                  Assert.assertEquals((int) version, 4);
```

Comment:
nit: Can you change the check to:
```
                if (colo.equals("dc-1")) {
                  Assert.assertEquals((int) version, 4);
                } else {
                  Assert.assertEquals((int) version, 3);
                }
```

This is to avoid the test from failing if the number of colos are increased